File: src/backend/src/model_detection/yolo_inference_service/lambda_handler.py
from __future__ import annotations

import logging
from typing import Any

logger = logging.getLogger("yolo_lambda_handler")

# Web-adapter se invoke `handler(event, context)` cho moi request.
# FastAPI khong can handler rieng (web-adapter goi truc tiep vao ASGI app),
# nhung Lambda runtime can mot symbol ten "handler" de CMD co the resolve.
#
# Tra ve mot response 200 mac dinh; web-adapter se forward HTTP request
# sang uvicorn dang chay o port 8000.


def handler(event: Any, context: Any) -> dict[str, Any]:
    logger.info("YOLO inference handler invoked event=%s", type(event).__name__)
    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": '{"service":"yolo-inference","status":"ok"}',
    }


async def async_handler(event: Any, context: Any) -> dict[str, Any]:
    return handler(event, context)

File: src/backend/src/lambda_handler.ts
/**
 * Lambda handler cho backend API (Express).
 *
 * aws-lambda-web-adapter doc HTTP request tu port 3000 (do express listen)
 * va tu forward den Lambda runtime. Handler nay chi la stub de Lambda runtime
 * resolve CMD, tra ve 200 OK.
 *
 * Neu muon khong dung web-adapter, wrap bang serverless-http:
 *     import serverless from 'serverless-http';
 *     import app from './app';
 *     export const handler = serverless(app);
 */
import type { Context } from './types/aws-lambda';

export const handler = async (event: any, context: Context) => {
  return {
    statusCode: 200,
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      service: 'cooksmart-backend-api',
      status: 'ok',
      message: 'Use web-adapter port 3000 for actual requests',
    }),
  };
};
